packages/stripe-integration/src/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -338,12 +328,15 @@ export default class StripeUPEPaymentStrategy implements PaymentStrategy {
         methodId: string,
     ) {
         const { containerId, style, render } = stripeupe;
-        const state = await this._store.dispatch(
-            this._paymentMethodActionCreator.loadPaymentMethod(gatewayId, {
-                params: { method: methodId },
-            }),
-        );
-        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        const state = await this.paymentIntegrationService.loadPaymentMethod(gatewayId, {
+            params: { method: methodId },
+        });
+        const paymentMethod = state.getPaymentMethodOrThrow(methodId);
+
+        if (!paymentMethod || !isStripeUPEPaymentMethodLike(paymentMethod)) {
```

Comment:
You are checking if payment method is existing inside type guard, I don't think that we need additional check there
```suggestion
        if (!isStripeUPEPaymentMethodLike(paymentMethod)) {
```